tests/test_range.py

```diff
@@ -103,6 +103,101 @@ def test_implicit(self):
                           datetime(2017,  6,  16, 11,  0,  0),
                           datetime(2017,  6,  16, 12,  0,  0))
 
+
+    def test_context_past_future(self):
+        # Current period
+        self.assert_range('2017',
+                          datetime(2017, 1, 1),
+                          datetime.now(),
+                          context=CONTEXT_PAST)
+
+        self.assert_range('2017',
+                          datetime.now(),
+                          datetime(2018, 1, 1),
+                          context=CONTEXT_FUTURE)
+
+        self.assert_range('june',
+                          datetime(2017, 6, 1),
+                          datetime.now(),
+                          context=CONTEXT_PAST)
+
+        self.assert_range('june',
+                          datetime.now(),
+                          datetime(2017, 7, 1),
+                          context=CONTEXT_FUTURE)
+
+        self.assert_range('2017 june',
+                          datetime(2017, 6, 1),
+                          datetime.now(),
+                          context=CONTEXT_PAST)
+
+        self.assert_range('2017 june',
+                          datetime.now(),
+                          datetime(2017, 7, 1),
+                          context=CONTEXT_FUTURE)
+
+        # A past period - no effect
+        self.assert_range('2000',
+                          datetime(2000, 1, 1),
+                          datetime(2001, 1, 1),
+                          context=CONTEXT_PAST)
+
+        self.assert_range('2000',
+                          datetime(2000, 1, 1),
+                          datetime(2001, 1, 1),
+                          context=CONTEXT_FUTURE)
+
+        self.assert_range('may',
```

Comment:
Looks like many of these might fail next January

Reply:
You mean they might fail in January 2018. Actually, the current date is hardcoded in the tests using the `freezegun` library, so it will never be 2018 inside the `RangeTest` class!